Source/WebCore/dom/Element.cpp

```diff
@@ -2298,19 +2298,31 @@ void Element::setElementsArrayAttribute(const QualifiedName& attributeName, std:
     }
 }
 
-void Element::classAttributeChanged(const AtomString& newClassString)
+void Element::classAttributeChanged(const AtomString& newClassString, AttributeModificationReason reason)
 {
     // Note: We'll need ElementData, but it doesn't have to be UniqueElementData.
     if (!elementData())
         ensureUniqueElementData();
 
-    {
+    ([&]() ALWAYS_INLINE_LAMBDA {
+        if (reason == AttributeModificationReason::Parser) {
+            // If ElementData is ShareableElementData created in parserSetAttributes,
+            // it is possible that SpaceSplitString is already created and set.
+            // We also do not need to invalidate caches / styles since it is not inserted to the tree yet.
+            if (elementData()->classNames().keyString() == newClassString)
```

Comment:
Should we assert here that `!elementData.isUnique()`?

Reply:
I think this is possible that elementData is no longer sharable because of the other attribute's copy, but still it has the same key string which is set when it was sharable before.